compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp

```diff
@@ -1219,17 +1213,11 @@ uptr GetPageSize() {
   int rv = internal_sysctl(mib, 2, &pz, &pzl, nullptr, 0);
   CHECK_EQ(rv, 0);
   return (uptr)pz;
+#    elif SANITIZER_ANDROID
+  // Using `getpagesize` because calling `getauxval` or `sysconf` from the
+  // `.preinit_array` can cause crashes on some older API levels.
+  return getpagesize();
```

Comment:
I don't think that actually works..  that calls through to `getauxval` via `page_size`: https://cs.android.com/android/platform/superproject/main/+/main:bionic/libc/platform/bionic/page.h;l=26?q=page_size%20f:bionic&sq=&ss=android%2Fplatform%2Fsuperproject%2Fmain

Reply:
HI. I don't know how `PAGE_SIZE` could become undefined?

It also seems to be hardcoded to `1 << 12`, which is `4096`.
https://cs.android.com/android/platform/superproject/main/+/main:prebuilts/gcc/linux-x86/host/x86_64-linux-glibc2.17-4.8/sysroot/usr/include/x86_64-linux-gnu/sys/user.h;drc=047082e7ae4aa2c0a066819b5f3a8307e2c52f0e;l=175

Anyway, in my device, it simply returns `PAGE_SIZE`.
```fish
libc.so`getpagesize:
->  0x7fb7cc8458 <+0>: orr    w0, wzr, #0x1000
    0x7fb7cc845c <+4>: ret  
```

Reply:
It seems that `PAGE_SIZE` is deprecated in Android.

https://android.googlesource.com/platform/bionic/+/master/libc/include/bits/page_size.h

Reply:
Yes, for new devices `PAGE_SIZE` should be undefined, to support different page sizes. Please revert to the original CL, sorry for the confusion Dan's comments caused.

Reply:
Oh ok, I only tested this on Android 8.1 using Termux, though.